src/core/pointcloud/qgspointcloudlayerelevationproperties.cpp

```diff
@@ -127,11 +128,28 @@ QgsDoubleRange QgsPointCloudLayerElevationProperties::calculateZRange( QgsMapLay
   {
     if ( pcLayer->dataProvider() )
     {
+      double zMin = std::numeric_limits<double>::quiet_NaN();
+      double zMax = std::numeric_limits<double>::quiet_NaN();
       const QgsPointCloudStatistics stats = pcLayer->statistics();
+      if ( !stats.statisticsMap().isEmpty() )
+      {
+        // try to fetch z range from provider metadata
+        zMin = stats.minimum( QStringLiteral( "Z" ) );
+        zMax = stats.maximum( QStringLiteral( "Z" ) );
+      }
+      // try to fetch the elevation properties from virtual point cloud metadata
+      else if ( QgsVirtualPointCloudProvider *virtualProvider = dynamic_cast< QgsVirtualPointCloudProvider * >( pcLayer->dataProvider() ) )
+      {
+        zMin = virtualProvider->subIndexes()[0].zRange().lower();
+        zMax = virtualProvider->subIndexes()[0].zRange().upper();
+        for ( QgsPointCloudSubIndex subIndex : virtualProvider->subIndexes() )
+        {
+          const QgsDoubleRange newRange = subIndex.zRange();
```

Comment:
While not very probable, `subIndex.zRange()` may be also a default constructed infinite range if the vpc has 4 coordinate instead of 6 coordinate bboxes.
These infinite range subindexes should be skipped or the zMin/zMax doubles will overflow after the multiplication later on.